scss/prototype/_spacing.scss

```diff
@@ -0,0 +1,204 @@
+// Foundation for Sites by ZURB
+// foundation.zurb.com
+// Licensed under MIT Open Source
+
+////
+/// @group prototype-spacing
+////
+
+/// Responsive breakpoints for spacing classes (margin and padding)
+/// @type Boolean
+$prototype-spacing-breakpoints: $global-prototype-breakpoints !default;
+
+/// Default number of spacers count (margin and padding)
+/// @type Number
+$prototype-spacers-count: 3 !default;
+
+/// Margin helper mixin, all the values are multiplied by `$global-margin` which by default is equal to `1rem`
+/// @param {Number} $top [null] - Margin Top
+/// @param {Number} $right [null] - Margin Right
+/// @param {Number} $bottom [null] - Margin Bottom
+/// @param {Number} $left [null] - Margin Left
+@mixin margin(
+  $top: null,
+  $right: null,
+  $bottom: null,
+  $left: null
+) {
+  @if $top != null {
+    margin-top: $top * $global-margin !important;
+  }
+  @if $right != null {
+    margin-right: $right * $global-margin !important;
+  }
+  @if $bottom != null {
+    margin-bottom: $bottom * $global-margin !important;
+  }
+  @if $left != null {
+    margin-left: $left * $global-margin !important;
+  }
+}
+
+/// Padding helper mixin, all the values are multiplied by `$global-padding` which by default is equal to `1rem`
+/// @param {Number} $top [null] - Padding Top
+/// @param {Number} $right [null] - Padding Right
+/// @param {Number} $bottom [null] - Padding Bottom
+/// @param {Number} $left [null] - Padding Left
+@mixin padding(
+  $top: null,
+  $right: null,
+  $bottom: null,
+  $left: null
+) {
+  @if $top != null {
+    padding-top: $top * $global-padding !important;
+  }
+  @if $right != null {
+    padding-right: $right * $global-padding !important;
+  }
+  @if $bottom != null {
+    padding-bottom: $bottom * $global-padding !important;
+  }
+  @if $left != null {
+    padding-left: $left * $global-padding !important;
+  }
+}
+
+@mixin foundation-prototype-spacing {
+  @for $spacer from 0 through $prototype-spacers-count {
+    // All Sides
+    .margin-#{$spacer} {
+      @include margin($spacer, $spacer, $spacer, $spacer);
+    }
+
+    .padding-#{$spacer} {
+      @include padding($spacer, $spacer, $spacer, $spacer);
+    }
+
+    // Top Side
+    .margin-top-#{$spacer} {
```

Comment:
Ohk no issue .... But if you ask me in any day i would like `mx0` instead of `margin-horizontal-0` for these!
Yes foundation convention is not `mx0` but then .... its prototype i guess (fast coding) ??